WalletWasabi/WabiSabi/Client/InternalDestinationProvider.cs

```diff
@@ -14,10 +14,21 @@ public InternalDestinationProvider(KeyManager keyManager)
 
 	private KeyManager KeyManager { get; }
 
-	public IEnumerable<IDestination> GetNextDestinations(int count)
+	public IEnumerable<IDestination> GetNextDestinations(int count, bool preferTaproot)
```

Comment:
Why is it called `preferTaproot`? It can give back non-taproot as well?

Reply:
Trezor would give back Taproot but the internal wallet should give back segwit? 
Below in the logic, it seems like if we prefer taproot but the wallet does not support it we won't give back anything.

Reply:
> Why is it called preferTaproot? It can give back non-taproot as well?

Yes. Wasabi is the one that has to use segwit **and** taproot scriptPubKeys because Trezor will use taproot only and BtcPayServer will be segwit only.

> Trezor would give back Taproot but the internal wallet should give back segwit?

No. Wasabi must support both script types for coinjoins.

> Below in the logic, it seems like if we prefer taproot but the wallet does not support it we won't give back anything.

There are no such a logic. Wasabi currently uses only segwit. Always. With no exception. And that is because that work was not finished and then I disabled it and the plan is to continue it in next PRs.

Reply:
ACK